### PR TITLE
Add `--rdg-scroll-height` variable

### PR DESCRIPTION
**Pull request by amanmahajan7**
This can be useful to set grid height based on the number of rows. For example we can do something like

```
block-size: 100%;
max-height: var(--rdg-scroll-height);
```

### CODE REVIEW

File: src/hooks/useGridDimensions.ts [hunk 1]
```diff
@@ -17,11 +18,13 @@ export function useGridDimensions() {
 
     const { clientWidth, clientHeight, offsetWidth, offsetHeight } = gridRef.current!;
     const { width, height } = gridRef.current!.getBoundingClientRect();
-    const initialWidth = width - offsetWidth + clientWidth;
+    const horizontalScrollbarWidth = offsetWidth - clientWidth;
```
Comment (nstepien) [inline src/hooks/useGridDimensions.ts:21]: Isn't that the vertical scrollbar width instead?

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -412,6 +413,8 @@ function DataGrid<R, SR, K extends Key>(
   const maxColIdx = columns.length - 1;
   const selectedCellIsWithinSelectionBounds = isCellWithinSelectionBounds(selectedPosition);
   const selectedCellIsWithinViewportBounds = isCellWithinViewportBounds(selectedPosition);
+  const rowsHeight =
+    headerRowHeight + totalRowHeight + summaryRowsHeight + horizontalScrollbarWidth;
```
Comment (nstepien) [inline src/DataGrid.tsx:417]: Why do we add the **vertical** scrollbar **width** to the height?

File: src/hooks/useGridDimensions.ts (hunk 1 above)
Comment (amanmahajan7) [reply src/hooks/useGridDimensions.ts:21]: 🤦‍♂️

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -1068,6 +1070,7 @@ function DataGrid<R, SR, K extends Key>(
           gridTemplateRows: templateRows,
           '--rdg-header-row-height': `${headerRowHeight}px`,
           '--rdg-summary-row-height': `${summaryRowHeight}px`,
+          '--rdg-rows-height': `${rowsHeight}px`,
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply src/DataGrid.tsx]: Or `--rdg-scroll-height`

[the remark this answers is not in the document]
Comment (nstepien) [reply src/DataGrid.tsx]: scroll height is good

File: src/hooks/useGridDimensions.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export function useGridDimensions() {
   const gridRef = useRef<HTMLDivElement>(null);
   const [inlineSize, setInlineSize] = useState(1);
   const [blockSize, setBlockSize] = useState(1);
+  const [horizontalScrollbarWidth, setHorizontalScrollbarWidth] = useState(0);
```
Comment (nstepien) [inline src/hooks/useGridDimensions.ts:10]: width -> height

**Review by nstepien — APPROVED** [review]

**Pull request merged**